In [3]:
import numpy as np
import pandas as pd 



In [6]:
df = pd.read_csv('~/Downloads/Integrify Assignement/bank-additional/bank-additional-full.csv', delimiter= ';')

In [7]:
df.head()

age        job  marital    education  default housing loan    contact  \
0   56  housemaid  married     basic.4y       no      no   no  telephone   
1   57   services  married  high.school  unknown      no   no  telephone   
2   37   services  married  high.school       no     yes   no  telephone   
3   40     admin.  married     basic.6y       no      no   no  telephone   
4   56   services  married  high.school       no      no  yes  telephone   

  month day_of_week  ...  campaign  pdays  previous     poutcome emp.var.rate  \
0   may         mon  ...         1    999         0  nonexistent          1.1   
1   may         mon  ...         1    999         0  nonexistent          1.1   
2   may         mon  ...         1    999         0  nonexistent          1.1   
3   may         mon  ...         1    999         0  nonexistent          1.1   
4   may         mon  ...         1    999         0  nonexistent          1.1   

   cons.price.idx  cons.conf.idx  euribor3m  nr.employed   y  
0          93.994          -36.4      4.857       5191.0  no  
1          93.994          -36.4      4.857       5191.0  no  
2          93.994          -36.4      4.857       5191.0  no  
3          93.994          -36.4      4.857       5191.0  no  
4          93.994          -36.4      4.857       5191.0  no  

[5 rows x 21 columns]

The data is related with direct marketing campaigns of a Portuguese banking institution. The marketing campaigns were based  
on phone calls. Often , more than one contact to the same client was required, in order to access if the product (bank term deposit)  
would be ('yes') or not('no') subscribed.  







The Data consist of 21 aatributes, where 20 attributes are input variables and 1 output variables (desired target). The   
attributes and their types are as follow:  
**Input Variable**

1. age = **numeric**
2. job = type of job, **categorical**
3. marital = marital status, **categorical**
4. education = **categorical**
5. default = has credit default?, **categorical**
6. housing = has housing loan?, **categorical**
7. loan = has personal loan?, **categorical**
8. contact = contact communication type, **categorical**
9. month = last contact month of year, **categorical**
10. day_of_week = last contact day of the week, **categorical**
11. duration = contact duration, **numeric**
12. campaign = number of contacts performed during the campaign, **numeric**
13. pdays = number of days passed by after the client was last contacted  from a previous campaign(999 means clieant was not contacted earlier), **numeric**     

14. previous = number of contacts performed before this campaign and for this client, **numeric**
15. poutcome = outcome of the previous marketing campaign, **categorical**
16. emp.var.rate = employment variation rate - quaterly indicator, **numeric**
17. cons.price.idx = consumer price index, monthly indicator, **numeric**
18. cons.conf.idx = consumer confidence index, monthly indicator, **numeric**
19. euribor3m = euribor 3 month rate, daily indicator, **numeric**
20. nr.employed = number of employees, quaterly indicator, **numeric**

**Output Variable**  
21. y = has the client subscribed a term deposit?, **categorical**


The main objective of this data project is to analyze the variables, descriptive statistics and  visualization of the data.

'C:\\Users\\SOBHAN\\Downloads\\Integrify notes\\python\\project sample'